plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/VacuumProcedure.java

```diff
@@ -300,17 +318,32 @@ private void doVacuum(
                 Location fileLocation = Location.of(location);
                 TrinoInputFile inputFile = fileSystem.newInputFile(fileLocation);
                 filesToDelete.add(inputFile);
+                filesToDeleteSize += inputFile.length();
+            }
+            long readVersion = handle.getReadVersion();
+            if (isVacuumLoggingEnabled) {
+                logVacuumStart(handle.location(), session, readVersion, filesToDelete.size(), filesToDeleteSize);
             }
             int totalFilesToDelete = filesToDelete.size();
             int batchCount = (int) Math.ceil((double) totalFilesToDelete / DELETE_BATCH_SIZE);
-            for (int batchNumber = 0; batchNumber < batchCount; batchNumber++) {
-                int start = batchNumber * DELETE_BATCH_SIZE;
-                int end = Math.min(start + DELETE_BATCH_SIZE, totalFilesToDelete);
+            String status = "FAILED";
+            try {
+                for (int batchNumber = 0; batchNumber < batchCount; batchNumber++) {
+                    int start = batchNumber * DELETE_BATCH_SIZE;
+                    int end = Math.min(start + DELETE_BATCH_SIZE, totalFilesToDelete);
 
-                List<TrinoInputFile> batch = filesToDelete.subList(start, end);
-                fileSystem.deleteFiles(batch.stream().map(TrinoInputFile::location).collect(toImmutableList()));
+                    List<TrinoInputFile> batch = filesToDelete.subList(start, end);
+                    numActualFilesDeleted += batch.size();
+                    fileSystem.deleteFiles(batch.stream().map(TrinoInputFile::location).collect(toImmutableList()));
+                }
+                status = "COMPLETED";
+            }
+            finally {
+                if (isVacuumLoggingEnabled) {
+                    int numVacuumedDirectories = Set.of(filesToDelete.stream().map(f -> f.location().parentDirectory())).size();
```

Comment:
numVacuumedDirectories can be wrong if we will fail in the middle of operation

Reply:
metrics are set to 0 in case of failure. See https://github.com/delta-io/delta/blob/68275d150db7980419836d9b891c61cddb986457/spark/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala#L448